### DISCUSSION TITLE
2011/05/22/%E5%8F%8D%E9%A6%88%E8%8A%82%E7%82%B9

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#328] [Announcements]
# 2011/05/22/%E5%8F%8D%E9%A6%88%E8%8A%82%E7%82%B9

https://labview.qizhen.xyz/

https://blog.qizhen.xyz/2011/05/22/%E5%8F%8D%E9%A6%88%E8%8A%82%E7%82%B9

**Comment by ruanqizhen** [comment]
之前博客下的留言：

jahnson
2012年06月23日 13:56 编辑
本博文中第五个VI，延迟3次后我实验的结果为2，而不是3了，请阮大侠再验证一下了；但是第六个VI的结果是9了，跟我验证的相符了，如果这样那后面3*3=9的解释好像就不正确了，没法传图上来，谅解。端午佳节快乐！

回复
ruanqizhen
2012年06月25日 09:37 编辑
这个地方我写错了，之后有人发现了错误，但我还一直没来得及改正。既然又被指出了，我今天就直接在原文上更新一下。

回复
ruanqizhen
2012年06月25日 09:45 编辑
改好了

回复
静
2011年05月31日 17:37 编辑
请教一下和这个主题不大相关的问题。可以讲一下关于用labview编写半峰宽或者峰面积方面的知识吗？ 半峰宽有两种计算方法：
一种是半高宽法fwhm，即做峰底的切线L，在峰高一半的地方做L平行线。一种是积分法，做峰底的切线L，测量峰的面积，测量峰的高度，
用“面积/高度”得到峰宽。我采用第二种方法，自己编写了一个VI，采用labview自带的积分模块，但是，上下限的动态确定有点问题，峰面积求取结果不对。您认为有什么方法解决吗？非常感谢

回复
ruanqizhen
2011年06月2日 08:59 编辑
我对信号处理方面没有了解，这个问题帮不了你了。

回复
Pingback： 《我和LabVIEW》补充内容和注释 « 凡人琐忆 编辑
ruanqizhen
2011年05月27日 13:06 编辑
删除数组需要重新排列数组元素位置，所以效率低。
LabVIEW的最初的目标是针对非专业程序员得，所以并没有透露很多细节给用户。现在LabVIEW这么庞大，可能没人弄得清所有底层的东西了。对于一些特定的问题，可能主要还是得靠编程者自己摸索。或者去论坛询问一下，集众人之力。

回复
huangchen
2011年05月27日 14:53 编辑
唉，个人认为如果底层用的链表的话，重排位置应该也不会太耗时间啊~呵呵~可能使用的那种连续的线性表吧~

回复
ruanqizhen
2011年05月27日 17:19 编辑
这个毕竟叫做“数组”，底层肯定也是真正的数组，否则索引就很慢了。

回复
huangchen
2011年06月2日 14:23 编辑
对，我2了，您说的很有道理，谢谢~

回复
huangchen
2011年05月27日 00:02 编辑
阮大侠，您好，我使用labview也有1年多了，最近做一个很小的功能的时候，发现拆分一维数组和删除数组元素这两个函数都可以完成一个一维数组的循环拆分（使用移位寄存器），但是第一个函数要比第二个函数效率高几十倍。

我想问一下，为什么拆分一维数组要比删除数组元素效率高这么多？我这里有测试vi，不过不知道怎么发给您；
第二，个人有时候感觉labview封装的挺好，用起来非常方便，但是却不能让我们比较容易的了解到它的底层，它的某些函数如何实现的，也就不能很方便地选取更适合的函数来开发程序（有时候选用不好就会导致程序效率很低下），您认为有解决这个问题的好办法么，或者有相应的书籍材料么？

回复
